pkg/skaffold/schema/versions.go

```diff
@@ -90,28 +91,24 @@ func ParseConfig(filename string, applyDefaults bool) (util.VersionedConfig, err
 	return cfg, nil
 }
 
-// CheckVersionIsLatest checks that a given version is the most recent.
-func CheckVersionIsLatest(apiVersion string) error {
-	parsedVersion, err := apiversion.Parse(apiVersion)
+// UpgradeToLatest upgrades a configuration to the latest version.
+func UpgradeToLatest(vc util.VersionedConfig) (util.VersionedConfig, error) {
+	var err error
+
+	// first, check to make sure config version isn't too new
+	version, err := apiversion.Parse(vc.GetVersion())
 	if err != nil {
-		return errors.Wrap(err, "parsing api version")
+		return nil, errors.Wrap(err, "parsing api version")
 	}
-
-	if parsedVersion.LT(apiversion.MustParse(latest.Version)) {
-		return errors.New("config version out of date: run `skaffold fix`")
+	latestVersion := apiversion.MustParse(latest.Version)
+	if version.GT(latestVersion) {
+		return nil, errors.New("config version is too new for this version of skaffold: upgrade skaffold")
 	}
 
-	if parsedVersion.GT(apiversion.MustParse(latest.Version)) {
-		return errors.New("config version is too new for this version of skaffold: upgrade skaffold")
+	if version.LT(latestVersion) {
+		logrus.Infof("config version %s out of date: upgrading to latest (%s)", vc.GetVersion(), latest.Version)
```

Comment:
Maybe we should print something out to stdout instead of just info to let the user know

Reply:
Its not in the diff but how about something slightly more idiomatic

```
if version == latest, return latest
if version GT latest, return error

upgrade
return upgraded
```